src/test/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerRunFingerprintFacetTest.java

```diff
@@ -46,14 +46,17 @@ public void test_readResolve() throws Exception {
         FreeStyleProject p = rule.createFreeStyleProject("test");
         FreeStyleBuild b = rule.assertBuildStatusSuccess(p.scheduleBuild2(0));
         
-        ContainerRecord r = new ContainerRecord("192.168.1.10", "cid", null, "magic", System.currentTimeMillis(), Collections.<String, String>emptyMap());
-        DockerFingerprints.addRunFacet(IMAGE_ID, r, b);
-        Fingerprint fingerprint = DockerFingerprints.of(IMAGE_ID);
-        DockerRunFingerprintFacet facet = (DockerRunFingerprintFacet) fingerprint.getFacets().iterator().next();
+        ContainerRecord r1 = new ContainerRecord("192.168.1.10", "cid", IMAGE_ID, "magic", System.currentTimeMillis(), Collections.<String, String>emptyMap());
+        DockerFingerprints.addRunFacet(r1, b);
 
-        Assert.assertNull(r.getImageId());
+        Fingerprint fingerprint = DockerFingerprints.of(IMAGE_ID);
+        DockerRunFingerprintFacet facet = new DockerRunFingerprintFacet(fingerprint, System.currentTimeMillis(), IMAGE_ID);
+        ContainerRecord r2 = new ContainerRecord("192.168.1.10", "cid", null, "magic", System.currentTimeMillis(), Collections.<String, String>emptyMap());
+        facet.add(r2);
+        
+        Assert.assertNull(r2.getImageId());
         facet.readResolve();
```

Comment:
Would still prefer a more natural test here, that after saving the record and running `rule.jenkins.reload()` you get everything back, but am not going to insist on it.